pysmt/oracles.py

```diff
@@ -223,14 +223,12 @@ def walk_lira(self, formula, args, **kwargs):
 
     def walk_times(self, formula, args, **kwargs):
         """Extends the Theory with Non-Linear, if needed."""
-        assert len(args) == 2
         theory_out = args[0]
         for t in args[1:]:
             theory_out = theory_out.combine(t)
-        # Check for non-linear
-        left, right = formula.args()
-        if len(left.get_free_variables()) != 0 and \
-           len(right.get_free_variables()) != 0:
+        # Check for non-linear counting the arguments having at least
+        # one free variable
+        if sum(1 for x in formula.args() if len(x.get_free_variables()) != 0) > 1:
```

Comment:
```if any(len(x.get_free_variables()) != 0):```
is not correct.

Here we need to check if at least two arguments are non-constant (hence I check if the count is > 1). Surely we can avoid the len, but I feel that leaving it is clearer.